src/lib/tuning/index/index_operation.cpp

```diff
@@ -0,0 +1,85 @@
+#include "index_operation.hpp"
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "sql/sql_query_cache.hpp"
+#include "sql/sql_query_plan.hpp"
+#include "storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp"
+#include "storage/index/group_key/composite_group_key_index.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+void IndexOperation::execute() {
+  if (_create) {
+    _create_index();
+  } else {
+    _delete_index();
+  }
+}
+
+void IndexOperation::print_on(std::ostream& output) const {
+  output << "IndexOperation{" << (_create ? "Create" : "Delete") << " on " << _column << "}";
+}
+
+void IndexOperation::_create_index() {
+  auto table = StorageManager::get().get_table(_column.table_name);
+  switch (_type) {
+    case ColumnIndexType::GroupKey:
+      table->create_index<GroupKeyIndex>(_column.column_ids);
+      break;
+    case ColumnIndexType::CompositeGroupKey:
+      table->create_index<CompositeGroupKeyIndex>(_column.column_ids);
+      break;
+    case ColumnIndexType::AdaptiveRadixTree:
+      table->create_index<AdaptiveRadixTreeIndex>(_column.column_ids);
+      break;
+    default:
+      Fail("Can not create invalid/unknown index type");
+  }
+
+  _invalidate_cache();
```

Comment:
Feel free to adapt the cache interface so that you can selectively remove entries. However, there might be multiple entries affected by this index creation decision.

Reply:
#756 addresses this point.

It seems to require a more detailed discussion than is possible within this scope.